proxy/envoy/discovery.go

```diff
@@ -302,6 +321,37 @@ func (ds *DiscoveryService) clearCache() {
 	ds.rdsCache.clear()
 }
 
+// ListAllEndpoints responds with all Services and is restricted to a single service key
```

Comment:
What do you mean by "and is restricted to a single service key"? Did you mean "is not restricted ..."?